cpp/include/cudf/io/nvcomp_adapter.hpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/utilities/export.hpp>
+
+#include <optional>
+#include <string>
+
+namespace CUDF_EXPORT cudf {
+namespace io::nvcomp {
+
+enum class compression_type { SNAPPY, ZSTD, DEFLATE, LZ4 };
+
+/**
+ * @brief Set of parameters that impact whether the use nvCOMP features is enabled.
+ *
+ */
+struct feature_status_parameters {
+  int lib_major_version;                 ///< major version
+  int lib_minor_version;                 ///< minor version
+  int lib_patch_version;                 ///< patch version
+  bool are_all_integrations_enabled;     ///< all integrations
+  bool are_stable_integrations_enabled;  ///< stable integrations
+  int compute_capability_major;          ///< cuda compute major version
+
+  /**
+   * @brief Default Constructor
+   */
+  feature_status_parameters();
+
+  /**pht
```

Comment:
Typo? Not sure what this means.
```suggestion
  /**
```